mixedbread_ai_langchain/retrievers/vector_store_retriever.py

```diff
@@ -0,0 +1,268 @@
+
+            if self.score_threshold is not None:
+                metadata["score_threshold"] = self.score_threshold
+
+            document = Document(page_content=page_content, metadata=metadata)
+            documents.append(document)
+
+        return documents
+
+    def _search_vector_stores(self, query: str) -> List[Document]:
+        """
+        Search vector stores and return documents.
+
+        Args:
+            query: Query string to search for.
+
+        Returns:
+            List of documents from search results.
+        """
+        try:
+            logger.debug(
+                f"Searching vector stores {self.vector_store_ids} with query: {query[:50]}..."
+            )
+
+            # Prepare request body for the new API structure
+            search_request = {
+                "query": query,
+                "vector_store_ids": self.vector_store_ids,
+                "top_k": self.top_k,
+            }
+
+            # Add optional parameters
+            if self.score_threshold is not None:
+                search_request["score_threshold"] = self.score_threshold
+            if self.return_metadata:
+                search_request["return_metadata"] = self.return_metadata
+
+            # Add any additional search options
+            search_request.update(self.search_options)
+
+            response: VectorStoreSearchResponse = (
+                self._client.client.vector_stores.search(**search_request)
+            )
+
+            documents = self._convert_search_results_to_documents(response)
+            logger.info(f"Vector store search returned {len(documents)} documents")
+            return documents
+
+        except Exception as e:
+            return self._handle_api_error(e, "vector store search", [])
+
+    def search_async(self, query: str):
+        """
+        Convenience method that returns an awaitable for async search.
+        For direct async operations, use retriever.aclient.vector_stores.search() instead.
+
+        Args:
+            query: Query string to search for.
+
+        Returns:
+            Awaitable that resolves to search response from the API.
+        """
+        # Async client is always available through MixedbreadClient
+        async_client = self._client.async_client
+
+        search_request = {
+            "query": query,
+            "vector_store_ids": self.vector_store_ids,
+            "top_k": self.top_k,
+        }
+
+        # Add optional parameters
+        if self.score_threshold is not None:
+            search_request["score_threshold"] = self.score_threshold
+        if self.return_metadata:
+            search_request["return_metadata"] = self.return_metadata
+
+        # Add any additional search options
+        search_request.update(self.search_options)
+
+        return async_client.vector_stores.search(**search_request)
+
+    def _get_relevant_documents(
+        self,
+        query: str,
+    ) -> List[Document]:
```

Comment:
```
    def _get_relevant_documents(
        self, query: str, *, run_manager: CallbackManagerForRetrieverRun
    ) -> list[Document]:
```
The signature seems to be missing the run_manager.
https://github.com/langchain-ai/langchain/blob/532e6455e93a31901af5fb071172fb12b9540563/libs/core/langchain_core/retrievers.py#L338-L340
The langchain docs seem no not be up to date?